lib/core/Stage.hpp

```diff
@@ -85,6 +97,9 @@ class Stage : public kotekanLogging {
 
     bufferContainer& buffer_container;
 
+    // A map to store all stat trackers in this stage.
+    std::map<std::string, std::shared_ptr<StatTracker>> stat_trackers;
```

Comment:
Likewise, I don't think we need to store this in the base class for every stage, just let each subclass store the copies of it's stat trackers as it sees fit.